addons/mail/models/mail_message.py

```diff
@@ -27,14 +27,16 @@ class Message(models.Model):
     _rec_name = 'record_name'
 
     @api.model
-    def _get_default_from(self):
-        if self.env.user.email:
-            return tools.formataddr((self.env.user.name, self.env.user.email))
-        raise UserError(_("Unable to post message, please configure the sender's email address."))
-
-    @api.model
-    def _get_default_author(self):
-        return self.env.user.partner_id
+    def default_get(self, fields):
+        res = super(Message, self).default_get(fields)
+        if ((not fields or 'email_from' in fields) and 'email_from' not in res) or \
```

Comment:
Would be nice to have a method in models.py to check that kind of stuff

Reply:
Looking at current definition it seems void fields is not supported anymore. Code simplified.